### ISSUE TITLE
max(0) false positive for cast_sign_loss

### ISSUE THREAD

**Issue by kornelski** [rust-lang/rust-clippy#11799]
### Summary

```rust
signed.max(0) as u64
```

shouldn't warn about loss of sign.

### Lint Name

cast_sign_loss

### Reproducer

I tried this code:

```rust
    let i = -1_i64;
    let _u = i.max(0) as u64;
```

I saw this happen:

```
warning: casting `i64` to `u64` may lose the sign of the value
```

I expected to see this happen:

- no warning

Same with `i.clamp(0, 999) as u64`.

### Version

```text
rustc 1.76.0-nightly (2b603f95a 2023-11-12)
binary: rustc
commit-hash: 2b603f95a48f10f931a61dd208fe3e5ffd64e491
commit-date: 2023-11-12
host: aarch64-apple-darwin
release: 1.76.0-nightly
LLVM version: 17.0.4
```
